shared/crypto/ed25519.go

```diff
@@ -119,6 +119,11 @@ func (priv *Ed25519PrivateKey) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return err
 	}
+	return priv.UnmarshalText([]byte(privateKey))
+}
+
+func (priv *Ed25519PrivateKey) UnmarshalText(data []byte) error {
```

Comment:
It says text but we pass in bytes and do an immediate cast to a string.

This implies that the byes are a UTF-8 representation of some text.

We need to change this to `UnmarshalString` and accept a `string` type. Or, be more explicit around the types and what's happening here (bytes, encoding, etc).

Reply:
I am substantially implementing https://pkg.go.dev/encoding#TextUnmarshaler and centralizing the logic (DRY).

Does it make sense?

Reply:
Yup